### PR TITLE
Custom resource to create deployment in api gateway

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -1297,6 +1297,59 @@ Resources:
           EVENT_BUS: !GetAtt ImportRequestsEventBus.Name
           IMPORT_EVENT_TYPE: import.cristin.entry-event
 
+  #============================ Deploy API =============================================================
```
Comment (axthosarouris) [inline template.yaml:1300]: Add a comment why is this necessary. It can be as simple as "This  solves the problem described here: http://....."

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Review by svenngjeto — DISMISSED** [review]

File: template.yaml [hunk 2]
```diff
@@ -1297,6 +1297,74 @@ Resources:
...
+
+  ApiGatewayCreateDeploymentCustomResource:
+    DependsOn: NvaPublicationApi
+    Type: AWS::CloudFormation::CustomResource
+    Properties:
+      ServiceToken: !GetAtt ApiGatewayCreateDeploymentLambda.Arn
+      RestApiId: !Ref NvaPublicationApi
+      StageName: !Ref NvaPublicationApi.Stage
+      Timestamp: '${BUILD_TIMESTAMP}'
+      AuthorizerArn: !Ref CognitoAuthorizerArn
```
Comment (janbj) [inline template.yaml:1367]: This will force update when authorizer ARN changes. Do we need both?

File: template.yaml [hunk 3]
```diff
@@ -1297,6 +1297,74 @@ Resources:
...
+            cfnresponse.send(event, context, responseStatus, responseData)
+
+  ApiGatewayCreateDeploymentCustomResource:
+    DependsOn: NvaPublicationApi
+    Type: AWS::CloudFormation::CustomResource
+    Properties:
+      ServiceToken: !GetAtt ApiGatewayCreateDeploymentLambda.Arn
+      RestApiId: !Ref NvaPublicationApi
+      StageName: !Ref NvaPublicationApi.Stage
+      Timestamp: '${BUILD_TIMESTAMP}'
```
Comment (janbj) [inline template.yaml:1366]: This will force update on every build.

File: template.yaml (hunk 2 above)
Comment (axthosarouris) [reply template.yaml:1367]: Keep  it for now.

**Review by axthosarouris — APPROVED** [review]

File: template.yaml (hunk 2 above)
Comment (svenngjeto) [reply template.yaml:1367]: It will cover the rare situation when we create a completely new Cognito without rebuilding publication-api?

**Review by svenngjeto — APPROVED** [review]
